planning/behavior_velocity_planner/src/utilization/util.cpp

```diff
@@ -593,34 +593,66 @@ LineString2d extendLine(
     {(p1 - length * t).x(), (p1 - length * t).y()}, {(p2 + length * t).x(), (p2 + length * t).y()}};
 }
 
-std::vector<lanelet::ConstLanelet> getLaneletsOnPath(
+boost::optional<int64_t> getNearestLaneId(
   const autoware_auto_planning_msgs::msg::PathWithLaneId & path,
-  const lanelet::LaneletMapPtr lanelet_map, const geometry_msgs::msg::Pose & current_pose)
+  const lanelet::LaneletMapPtr lanelet_map, const geometry_msgs::msg::Pose & current_pose,
+  boost::optional<size_t> & nearest_segment_idx)
 {
-  std::set<int64_t> unique_lane_ids;
-  auto nearest_segment_idx = tier4_autoware_utils::findNearestSegmentIndex(
+  boost::optional<int64_t> nearest_lane_id;
+
+  nearest_segment_idx = tier4_autoware_utils::findNearestSegmentIndex(
     path.points, current_pose, std::numeric_limits<double>::max(), M_PI_2);
 
-  // Add current lane id
+  if (!nearest_segment_idx) {
+    return boost::none;
+  }
+
   lanelet::ConstLanelets current_lanes;
   if (
     lanelet::utils::query::getCurrentLanelets(
       lanelet::utils::query::laneletLayer(lanelet_map), current_pose, &current_lanes) &&
     nearest_segment_idx) {
     for (const auto & ll : current_lanes) {
-      if (
-        ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0) ||
-        ll.id() == path.points.at(*nearest_segment_idx + 1).lane_ids.at(0)) {
-        unique_lane_ids.insert(ll.id());
+      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
+        *nearest_lane_id = ll.id();
+        return nearest_lane_id;
+      }
+    }
+
+    // if the lane_id of nearest_segment_idx does not involved in current_lanes,
+    // search the lane_id of nearest_segment_idx + 1
+    *nearest_segment_idx += 1;
+    for (const auto & ll : current_lanes) {
+      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
+        *nearest_lane_id = ll.id();
```

Comment:
```suggestion
        nearest_lane_id = ll.id();
```

Reply:
This is going to fix planner dies below case
![image](https://user-images.githubusercontent.com/65527974/178652104-8d037524-e8a9-4e2b-a22b-a0e01283aa0f.png)